src/Cli/dotnet/NugetPackageDownloader/NuGetPackageDownloader.cs

```diff
@@ -65,7 +69,9 @@ public NuGetPackageDownloader(
             _restoreActionConfig = restoreActionConfig ?? new RestoreActionConfig();
             _retryTimer = timer;
             _sourceRepositories = new();
-            _verifySignatures = verifySignatures;
+            // If windows or env variable is set, verify signatures
+            _verifySignatures = verifySignatures && (OperatingSystem.IsWindows() ? true 
+                : string.Equals(Environment.GetEnvironmentVariable(NuGetSignatureVerificationEnabler.DotNetNuGetSignatureVerification), bool.TrueString, StringComparison.OrdinalIgnoreCase) ?? OperatingSystem.IsLinux());
```

Comment:
```suggestion
                : string.Equals(Environment.GetEnvironmentVariable(NuGetSignatureVerificationEnabler.DotNetNuGetSignatureVerification), bool.FalseString, StringComparison.OrdinalIgnoreCase) ?  false : OperatingSystem.IsLinux());
```

Reply:
?? Can't be used on a boolean value like that in C#. The best thing to do imo is check if its explicitly disabled. If it's not explicitly disabled, then check if its linux, so that its enabled by default on linux and not enabled by default on Mac. The windows case is already handled.

Reply:
That isn't quite correct, though. And Environment.GetEnvironmentVariable returns a bool?